src/welcome/ui/components/WordInput.tsx

```diff
@@ -0,0 +1,51 @@
+import Grid from '@mui/material/Grid'
+import Autocomplete from '@mui/material/Autocomplete'
+import TextField from '@mui/material/TextField'
+
+interface Props {
+  dictionary: string[]
+  wordIdx: number
+  value: string
+  onChange: (wordIdx: number, value: string) => void
+  isCorrect: boolean
+}
+
+export default function WordInput({
+  dictionary,
+  wordIdx,
+  value,
+  onChange,
+  isCorrect,
+}: Props) {
+  const filter = (
```

Comment:
The custom `filter` function is just to prevent auto-completion from being triggered if the input has less than 2 characters, otherwise the list of options is too long.